src/resources/rmd/execute.R

```diff
@@ -491,6 +494,42 @@ apply_slides_patch <- function(includes) {
   includes
 }
 
+
+apply_responsive_patch <- function(format) {
+  if (isTRUE(format$metadata[["fig-responsive"]])) {
+
+    # tweak sizing for htmlwidget figures (use 100% to be responsive)
+    if (requireNamespace("htmlwidgets", quietly = TRUE)) {
+      htmlwidgets_resolveSizing <- htmlwidgets:::resolveSizing
+      size_options <- c("fig.height", "fig.width")
+      default_knitr_options <-  knitr::opts_chunk$get()[size_options]
+      
+      resolveSizing <- function(x, sp, standalone, knitrOptions = NULL) {
+          # default sizing resolution
+          sizing <- htmlwidgets_resolveSizing(x, sp, standalone, knitrOptions)
+          
+          # Do not alter if user has change fig.height or fig.width option
+          if (!is.null(knitrOptions) && !identical(default_knitr_options, knitrOptions[size_options])) {
+            return(sizing)
+          }
```

Comment:
This will do it per chunk. This means that if the `fig.*` options are changed on a specific chunk, it will not be modified but all the other chunk would have the transformation. 

If option is change globally `knitr::opts_chunk$set`, then all chunk would not be responsive. 

However, it is not perfect because `knitr::opts_chunk$get` will only get us the default knitr option, but not the default for the format that can be changed by Quarto. 

We could probably remove this and the two other related line above and do that later;